sormas-backend/src/main/java/de/symeda/sormas/backend/sormas2sormas/Sormas2SormasEncryptionService.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2020 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ *******************************************************************************/
+package de.symeda.sormas.backend.sormas2sormas;
+
+import java.io.FileInputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+
+import javax.ejb.EJB;
+import javax.ejb.LocalBean;
+import javax.ejb.Stateless;
+
+import com.google.common.collect.Lists;
+
+import de.symeda.sormas.api.Sormas2SormasConfig;
+import de.symeda.sormas.backend.common.ConfigFacadeEjb;
+import de.symeda.sormas.backend.crypt.CmsCreator;
+import de.symeda.sormas.backend.crypt.CmsReader;
+
+@Stateless
+@LocalBean
+public class Sormas2SormasEncryptionService {
+
+	@EJB
+	private ConfigFacadeEjb.ConfigFacadeEjbLocal configFacade;
+
+	public String encrypt(String data, String instanceID) throws Exception {
```

Comment:
Should the service encrypt byte arrays instead?

Reply:
I talked with Levi and he said that for the integration he would prefer Strings

Reply:
Then we have to rely on e.g. Base64. I'm just not sure if this is the right layer for that.
Also, the service should propably better accept the data as byte array, so that the encoding is left to the caller. This also allows for supporting formats that are not string-based.

Reply:
It makes sense, I replaced String with byte[]